client/src/pages/CoursePage.tsx

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useState } from 'react';
+import { Link, useParams } from 'react-router-dom';
+
+import { CourseInfo } from '../components/CourseInfo';
+import { CourseRequirements } from '../components/CourseRequirements';
+import { CourseReview } from '../components/CourseReview';
+import { CourseReviewPrompt } from '../components/CourseReviewPrompt';
+import { Layout } from '../components/Layout';
+import { useAuth } from '../hooks/useAuth';
+import { fetchClient } from '../lib/fetchClient';
+import { Course } from '../model/course';
+import { Requirements } from '../model/requirements';
+import { Review } from '../model/review';
+
+export const CoursePage = () => {
+  const params = useParams<{ id: string }>();
+  const [course, setCourse] = useState<Course>();
+  const [reviews, setReviews] = useState<Review[]>();
+  const user = useAuth();
+
+  useEffect(() => {
+    fetchClient
+      .getData<Course>(`/courses/${params.id?.toUpperCase()}`)
+      .then((data) => setCourse(data))
+      .catch((err) => console.log(err));
+    fetchClient
+      .getData<Review[]>(`/reviews?course_id=${params.id}`)
+      .then((data) => {
+        setReviews(data);
+      });
+  }, [params.id]);
+
+  if (course === null) {
+    return <div>404 not found</div>;
+  }
+
+  if (course === undefined || reviews === undefined) {
+    return <div>Loading...</div>; // TODO: some spinning comonent
```

Comment:
We have a spinner -> components/Spinner.tsx

Reply:
ill get on that